x/ccv/provider/keeper/distribution.go

```diff
@@ -191,7 +191,17 @@ func (k Keeper) AllocateTokensToConsumerValidators(
 	}
 
 	// Allocate tokens by iterating over the consumer validators
-	for _, consumerVal := range k.GetConsumerValSet(ctx, chainID) {
+	consumerVals, err := k.GetConsumerValSet(ctx, chainID)
+	if err != nil {
+		k.Logger(ctx).Error(
+			"cannot get consumer validator set while allocating rewards from consumer chain",
```

Comment:
```suggestion
			"cannot get consumer validator set while allocating rewards for consumer chain",
```

Reply:
the rewards are indeed also *from* a consumer chain, right? as in, they come from the consumer chain